lib/BitrateTest.ts

```diff
@@ -172,20 +185,31 @@ export class BitrateTest extends EventEmitter {
       this._setupDataChannel();
       this._startTest();
     });
+
+    this._timeout = setTimeout(() => {
+      const error: TimeoutError = new TimeoutError(
+        'Network timeout, exceeded limit of `${BITRATE_TEST_TIMEOUT_MS}`ms');
+      this._errors.push(error);
+      this.emit(BitrateTest.Events.Error, error);
+      this.stop();
+    }, BITRATE_TEST_TIMEOUT_MS);
```

Comment:
We have a private method `this._onError` that does things very similar to this. We should probably update it to also support TimeoutError or other types of DiagnosticsError and use that method here.